src/Microsoft.AspNetCore.Razor.Language/RazorProjectEngine.cs

```diff
@@ -37,54 +74,90 @@ public abstract class RazorProjectEngine
 
             var builder = new DefaultRazorProjectEngineBuilder(configuration, fileSystem);
 
-            AddDefaults(builder);
-
-            if (configuration.DesignTime)
-            {
-                AddDesignTimeDefaults(builder);
-            }
-            else
-            {
-                AddRuntimeDefaults(builder);
-            }
+            RazorEngine.AddDefaultPhases(builder.Phases);
+            AddDefaultsFeatures(builder.Features);
 
             configure?.Invoke(builder);
 
             return builder.Build();
         }
-
-        private static void AddDefaults(RazorProjectEngineBuilder builder)
+        
+        private static void AddDefaultsFeatures(ICollection<IRazorFeature> features)
```

Comment:
I decided to copy the existing code since this was diverging from the `RazorEngine` path more and more.